pkg/client/apiutil/apimachinery.go

```diff
@@ -169,3 +170,25 @@ func createRestConfig(gvk schema.GroupVersionKind, isUnstructured bool, baseConf
 
 	return cfg
 }
+
+// BoolPointerFlagFunc returns a function to be associated with the custom bool flag func
+//
+// Example:
+//
+// var fooPtr *bool
+// flag.Func("foo", "foo usage", apiutil.BoolPointerFlagFunc(fooPtr))
+func BoolPointerFlagFunc(ptr *bool) func(flagVal string) error {
+	return func(flagVal string) error {
+		// flagVal is empty, consider flag as not passed
+		if flagVal == "" {
+			return nil
+		}
+
+		boolVal, err := strconv.ParseBool(flagVal)
+		if err != nil {
+			return err
+		}
+		ptr = &boolVal
```

Comment:
Actually, this doesn't do what you think it does. The `ptr` variable holds a pointer and is scoped to this function. You will need to:
* Set its value (which is problematic here I think, because then it will always be non-nil?)
* Use a double pointer `**bool`
/lgtm cancel

Reply:
Okay, got it. Let me think and work on it. Thanks for pointing that out